src/common/tld.js

```diff
@@ -4,7 +4,19 @@ import { getDomain as getDomain_, getPublicSuffix as getPublicSuffix_ } from 'tl
  * tldts does not respect the public suffix list by default, but can be opt in manually
  * with the option `allowPrivateDomains`. Hoist the `sharedOpts` can also help avoid
  * re-creating the object every time.
+ *
+ * Note `extractHostname` and `validateHostname` are set to false because the inputs are
+ * from `new URL(url).hostname` and are known to be valid
  */
-const sharedOpts = { allowPrivateDomains: true };
-export const getDomain = url => getDomain_(url, sharedOpts);
-export const getPublicSuffix = url => getPublicSuffix_(url, sharedOpts);
+const getDomainSharedOpts = {
+  allowPrivateDomains: true,
+  extractHostname: false, // inputs are already hostnames
+  validateHostname: false, // inputs are already valid, no need to perform extra validation
+}
+
+const getPublicSuffixSharedOpts = {
+  allowPrivateDomains: true
+};
+
+export const getDomain = url => getDomain_(url, getDomainSharedOpts);
+export const getPublicSuffix = url => getPublicSuffix_(url, getPublicSuffixSharedOpts);
```

Comment:
AFAICT the same options can be used for both functions.

Reply:
I am worrying about the `getPublicSuffix` case here:

<img width="522" alt="image" src="https://github.com/violentmonkey/violentmonkey/assets/40715044/9c702ae7-a6aa-4f28-a114-333adcb33d24">

<img width="1104" alt="image" src="https://github.com/violentmonkey/violentmonkey/assets/40715044/665bcd5e-edda-46d9-b9ac-bebae63984b3">

The input of `getPublicSuffix` can be invalid IMHO.

Reply:
AFAICT it can't be invalid because it's extracted as dot-separated valid sequence via `((?:\.[-\w]+)+)/` or `(|(?:\.[-\w]+)+)`

Reply:
@tophf Exactly! Using regex can not ensure whether the input is a valid domain or not, thus extra check from `tldts` is required!

Also, the option will affect how `tldts` handles the invalid TLD:

https://runkit.com/sukkaw/652d2ea9557abc00083e0209

<img width="936" alt="image" src="https://github.com/violentmonkey/violentmonkey/assets/40715044/57283adf-ff08-4fdb-bb24-7826362813d2">

Reply:
No, my point is that our regex is guaranteed to extract a valid sequence.

Reply:
Also note that optimizing getDomain is pointless performance-wise as we use it only when showing the popup and when the user creates a new script for the tab.